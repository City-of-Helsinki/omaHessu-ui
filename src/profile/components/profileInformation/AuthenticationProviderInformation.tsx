import React from 'react';
import { useTranslation } from 'react-i18next';

import useProfile from '../../../auth/useProfile';
import LabeledValue from '../../../common/labeledValue/LabeledValue';
import { getAmrStatic } from './authenticationProviderUtil';
import styles from './authenticationProviderInformation.module.css';

function AuthenticationProviderInformation(): React.ReactElement | null {
  const { t } = useTranslation();
  const { profile } = useProfile();

  const amr = getAmrStatic(profile);

  if (!amr) {
    return null;
  }

  const authenticationMethodReferenceName = t(`identityProvider.${amr}`);

  return (
    <div className={styles['container']}>
      <div className={styles['label-section']}>
        <LabeledValue
          label={t('profileInformation.authenticationMethod')}
          value={authenticationMethodReferenceName}
        />
      </div>
    </div>
  );
}

export default AuthenticationProviderInformation;
