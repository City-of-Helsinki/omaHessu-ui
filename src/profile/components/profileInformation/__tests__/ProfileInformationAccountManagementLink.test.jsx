import React from 'react';
import { mount } from 'enzyme';
import enzymeToJson from 'enzyme-to-json';

import AuthenticationProviderInformation from '../AuthenticationProviderInformation';
import { mockUserCreator } from '../../../../common/test/userMocking';

let mockCurrentAmr;
const helsinkiAccountAMR = 'helusername-test';

jest.mock('../../../../config', () => ({
  helsinkiAccountAMR,
}));

jest.mock('../../../../auth/useProfile', () => () => mockUserCreator());

jest.mock('../authenticationProviderUtil', () => ({
  ...jest.requireActual('../authenticationProviderUtil'),
  getAmrStatic: () => mockCurrentAmr,
}));

describe('<ProfileInformationAuthenticationSourceBackLink /> ', () => {
  const defaultProps = {};
  const getWrapper = props =>
    mount(<AuthenticationProviderInformation {...defaultProps} {...props} />);
  describe('renders correctly when AMR is helsinkiAccountAMR', () => {
    beforeAll(() => {
      window._env_.REACT_APP_HELSINKI_ACCOUNT_AMR = helsinkiAccountAMR;
    });

    afterAll(() => {
      window._env_.REACT_APP_HELSINKI_ACCOUNT_AMR = 'helusername';
    });

    it('should render helsinki account link as expected based on config', () => {
      mockCurrentAmr = 'helsinkiAccount';
      const wrapper = getWrapper();
      expect(enzymeToJson(wrapper)).toMatchSnapshot();
    });
  });
  describe('renders correctly when AMR is tunnistusSuomifiAMR', () => {
    it('should render suomi.fi link as expected based on config', () => {
      mockCurrentAmr = 'tunnistusSuomifi';
      const wrapper = getWrapper();
      expect(enzymeToJson(wrapper)).toMatchSnapshot();
    });
  });
});
