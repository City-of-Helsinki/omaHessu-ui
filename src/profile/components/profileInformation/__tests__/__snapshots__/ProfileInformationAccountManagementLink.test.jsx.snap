// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileInformationAuthenticationSourceBackLink /> should render helsinki account link as expected based on config 1`] = `
<ProfileInformationAccountManagementLink>
  <div
    className="container"
  >
    <div
      className="labelSection"
    >
      <LabeledValue
        label="profileInformation.authenticationMethod"
        value="identityProvider.helsinkiAccount"
      >
        <div
          className="wrapper"
        >
          <strong
            className="label"
          >
            profileInformation.authenticationMethod
          </strong>
          <span
            className="value"
          >
            identityProvider.helsinkiAccount
          </span>
        </div>
      </LabeledValue>
    </div>
    <div
      className="link"
    >
      <a
        href="https://test-url"
        rel="noopener noreferrer"
        target="_blank"
      >
        profileInformation.doGoToAccountManagement
         
        <IconAngleRight>
          <svg
            className="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
            style={Object {}}
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
              fillRule="evenodd"
            >
              <path
                d="M0 24V0h24v24z"
              />
              <path
                d="M13 12.5l-5-5L9.5 6l6.5 6.5L9.5 19 8 17.5z"
                fill="currentColor"
              />
            </g>
          </svg>
        </IconAngleRight>
      </a>
    </div>
  </div>
</ProfileInformationAccountManagementLink>
`;
